include/arch/x86/linker.ld

```diff
@@ -136,6 +136,17 @@ SECTIONS
 
 	__data_ram_end = .;
 
+#ifdef CONFIG_FIRMWARE_VERSION_STRING
+	SECTION_PROLOGUE(_VERSION_SECTION_NAME, (OPTIONAL),)
+	{
+	. = ALIGN(0x40033FC0);
```

Comment:
don't do this!!

You are setting the location counter to the next address on a 0x40033FC0 byte boundary. This is wrong for three reasons

1) This is a hack. There is already a way to declare a fixed load address for a section, just google "Fixed load address for linker section" to get the syntax

2) Due to the size of flash, ALIGN(0x40033FC0) *will* coincidentally set the location counter to exactly 0x40033FC0, but it is entirely dependent on the size of flash (i.e. you might get 0x0, or 0x80067F80 depending on the size of flash)

3) You're setting the location counter to 0x40033FC0. So all "romable" sections after this one will be after 0x40033FC0..... meaning we'll probably lose a bunch of stuff